src/components/MAPI/clusters/ClusterList/ClusterListItem.tsx

```diff
@@ -144,41 +144,15 @@ const ClusterListItem: React.FC<IClusterListItemProps> = ({
   const clientFactory = useHttpClientFactory();
   const auth = useAuthProvider();
 
-  const orgNamespace = organization?.namespace;
-
-  const { canList: canListNpInOrgNamespace, canGet: canGetNpInOrgNamespace } =
-    usePermissionsForNodePools(provider, orgNamespace ?? '');
-
-  const hasReadPermissionsForNpInOrgNamespace =
-    canListNpInOrgNamespace && canGetNpInOrgNamespace;
-
-  const {
-    canList: canListNpInDefaultNamespace,
-    canGet: canGetNpInDefaultNamespace,
-  } = usePermissionsForNodePools(provider, 'default');
-
-  const hasReadPermissionsForNpInDefaultNamespace =
-    canListNpInDefaultNamespace && canGetNpInDefaultNamespace;
-
-  const nodePoolListForClusterKey = useMemo(() => {
-    switch (true) {
-      case hasReadPermissionsForNpInDefaultNamespace:
-        return fetchNodePoolListForClusterKey(cluster);
-      case hasReadPermissionsForNpInOrgNamespace:
-        return fetchNodePoolListForClusterKey(cluster, orgNamespace);
-      default:
-        return null;
-    }
-  }, [
-    cluster,
-    hasReadPermissionsForNpInDefaultNamespace,
-    hasReadPermissionsForNpInOrgNamespace,
-    orgNamespace,
-  ]);
+  const { canList: canListNodePools, canGet: canGetNodePools } =
```

Comment:
This refactors how we determine permissions for node pool resources: we check the user's permissions for node pools in the same namespace as the cluster's namespace.

e.g. If the cluster is in the `org-giantswarm` namespace, we check if the user has permissions for node pool resources in the `org-giantswarm` namespace.